rust/core-lib/src/edit_types.rs

```diff
@@ -0,0 +1,181 @@
+// Copyright 2018 Google Inc. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+//! A bunch of boilerplate for converting the `EditNotification`s we receive
+//! from the client into the events we use internally.
+//!
+//! This simplifies code elsewhere, and makes it easier to route events to
+//! the editor or view as appropriate.
+
+use movement::Movement;
+use ::rpc::{GestureType, LineRange, EditNotification, MouseAction};
+
+
+/// Events that only modify view state
+pub (crate) enum ViewEvent {
```

Comment:
Extremely minor formatting nit, most sources don't indicate a space between `pub` and `(crate)`